### PR TITLE
Opt-in init to speed up startup times

### PR DESCRIPTION
**Pull request by kingces95**

### CODE REVIEW

**Comment by rmarinho** [comment]
Can we have a better description on the PR @kingces95  ?

File: Xamarin.Forms.Core/Registrar.cs [hunk 1]
```diff
@@ -265,21 +266,21 @@ public static void RegisterRenderers(HandlerAttribute[] attributes)
...
 		public static void RegisterStylesheets()
 		{
-			var assembly = typeof(StyleSheets.StylePropertyAttribute).GetTypeInfo().Assembly;
+			var assembly = typeof(StylePropertyAttribute).GetTypeInfo().Assembly;
 
 #if NETSTANDARD2_0
-			object[] styleAttributes = assembly.GetCustomAttributes(typeof(StyleSheets.StylePropertyAttribute), true);
+			object[] styleAttributes = assembly.GetCustomAttributes(typeof(StylePropertyAttribute), true);
 #else
 			object[] styleAttributes = assembly.GetCustomAttributes(typeof(StyleSheets.StylePropertyAttribute)).ToArray();
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Core/Registrar.cs]: A small detail, after adding the namespace, `StyleSheets` is no longer necessary here.

`object[] styleAttributes = assembly.GetCustomAttributes(typeof(StylePropertyAttribute)).ToArray();`

File: Xamarin.Forms.Core/Registrar.cs [hunk 2]
```diff
@@ -340,13 +341,9 @@ public static void RegisterAll(Type[] attrTypes, ActivationFlags flags)
...
 
-					var length = attributes.Length;
-					for (var i = 0; i < length; i++)
-					{
-						var attribute = (HandlerAttribute)attributes[i];
-						if (attribute.ShouldRegister())
-							Registered.Register(attribute.HandlerType, attribute.TargetType, attribute.SupportedVisuals);
-					}
+					var handlerAttributes = new HandlerAttribute[attributes.Length];
+					Array.Copy(attributes, handlerAttributes, attributes.Length);
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Core/Registrar.cs]: :thumbsup:

**Comment by kingces95** [comment]
This change will not result in dramatic performance improvements. Instead, it was used to demonstrate the feasibility of building our profiler and using that profiler to identify a performance issue and to verify the fix.

I plan to rename some of the items published in this PR (if I can get away with that) when I push the other performance fixes that, taken together, will produce a startup performance improvement worth promoting. 

Those fixes are been developed and are now going through our PR process.

**Pull request merged**